src/api/Main/BuisnessLayer/Shrooms.Domain/Services/Kudos/KudosService.cs

```diff
@@ -459,12 +459,36 @@ public decimal[] GetMonthlyKudosStatistics(string id)
 
         public void AddKudosLog(AddKudosLogDTO kudosLog)
         {
-            AddKudosRequest(kudosLog);
+            if (UserHasPermission(kudosLog))
+            {
+                AddKudosRequest(kudosLog);
+            }
         }
 
         public void AddKudosLog(AddKudosLogDTO kudosDto, decimal points)
         {
-            AddKudosRequest(kudosDto, points);
+            if (UserHasPermission(kudosDto))
+            {
+                AddKudosRequest(kudosDto, points);
+            }
+        }
+
+        private bool UserHasPermission(AddKudosLogDTO kudosDto)
+        {
+            var authorizedRoles = new List<string>() { Constants.Authorization.Roles.Admin, Constants.Authorization.Roles.KudosAdmin };
```

Comment:
Minor code quality issue (can be fixed later):
- `authorizedRoles` initialuzation line could be put inside "if" statement
- `if (!kudosDto.IsActive)` could be inverted to reduce nesting